### PR TITLE
Implement delete functionality for delete button on the operations menu

### LINKED ISSUE #149: Provide UI for users to delete input
### Acceptance criteria
- Implement the confirmation dialog exactly as shown in the [mockup](https://www.figma.com/design/9QKsvDQ9jivgA9XfS0oqPd/Design-System---Linkta?node-id=2453-7935&t=Rwnz7IlUEgIWybky-0)
  - Position the component in the location specified in the mockup
  - Use the correct colors, fonts, and dimensions as detailed in the design specs
  - Implement all interactive states (e.g., hover, active, disabled) as per the mockup
  - Ensure the component's responsiveness matches the mockup across different screen sizes
- Confirmation dialog is shown before deleting a LinktaFlow.
- Ensure accessibility.
- Error handling with SnackBarNotification is in place
- React Query setup in #253 is integrated with the component

### LINKED ISSUE #265: Implement delete functionality for delete button on the operations menu
**Related Epic**
#120 

**Parent Ticket**
#149

**Acceptance Criteria**
- User is able to delete a linktaflow when clicking the delete button on the operations menu

### PR DESCRIPTION
**Pull request by AaronDreyfuss**
## Description
Implements functionality to the delete buttons associated with each user input in the sidebar.

Fixes #265

## Type of change

- [ ]  🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨  New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] Other 🗒️ ______________

## How Can this be tested? Testing Plan to review this PR

Run the application, navigate to the sidebar and click the delete button next to each user input in the sidebar to remove them. Verify the changes by checking the MongoDB database to confirm that each deleted item is removed. Observe the UI to ensure that the user input is immediately removed from the sidebar upon deletion and that the remaining inputs remain functional without any unintended changes.

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [x] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [x] I have ensured that my pull request title is descriptive.

### CODE REVIEW

**Comment by kinskiwu** [comment]
@AaronDreyfuss looks like the linting failed, pls fix the issue the push your changes

**Comment by kinskiwu** [comment]
Thanks for the good work! 

Please add a UI with styling to remind the user before they delete the input (see hi-fi mockups for linktaflow): https://www.figma.com/design/9QKsvDQ9jivgA9XfS0oqPd/Design-System---Linkta?node-id=2453-7935&t=WssUJOosegOpdOIS-0. 

For state management for userinputs, we recommend using the userinput store that reland set up with zustand. 

If you prefer to work on the other tasks in separate PRs, please create new issues in the github project and link them to this PR and the original issue. Ty!

**Comment by AaronDreyfuss** [comment]
> @AaronDreyfuss looks like the linting failed, pls fix the issue the push your changes

I fixed the linting issue

**Comment by kinskiwu** [comment]
@AaronDreyfuss I created a subtask for this PR and I updated the subject of this PR to better reflect the update.

**Review by kinskiwu — APPROVED** [review]

File: client/components/layout/UserInputList.tsx [hunk 1]
```diff
@@ -48,13 +50,30 @@ const UserInputList: React.FC<UserInputListProps> = ({
...
+            (input) =>
+              `${input._id}-${inputs.indexOf(input)}` !== selectedUserInputId,
+          ),
+        );
+      } catch (error) {
+        // console.error('Failed to delete user input:', error);
+      }
+    }
+    handleMenuClose();
+  }, [selectedUserInputId, handleMenuClose]);
```
Comment (colinvandergraaf) [inline client/components/layout/UserInputList.tsx:68]: super minor side note here that doesn't seem to impact functionality, but I get an eslint warning that wants inputs in this dependency array.  I think it makes sense, but everything is running fine.

**Review by colinvandergraaf — APPROVED** [review]
LGTM 🤘!

**Pull request merged**